subsys/net/ip/net_context.c

```diff
@@ -816,6 +816,18 @@ NET_CONN_CB(tcp_established)
 				     sys_get_be32(NET_TCP_HDR(pkt)->ack));
 	}
 
+	if ((s32_t)(sys_get_be32(NET_TCP_HDR(pkt)->seq) - context->tcp->send_ack) < 0) {
```

Comment:
What happens with this current check if the send_ack has wrapped in which case the check will give incorrect answer.
I wonder if it would be good idea at this point to fix the sequence check issue you mentioned in https://jira.zephyrproject.org/browse/ZEP-2120
So we could have a generic function that would check if the seq is properly greater than some other seq, and then call that function here.